includes/wikia/services/TemplateClassificationService.class.php

```diff
@@ -164,7 +165,13 @@ private function createApiClient() {
 		global $wgConsulUrl, $wgConsulServiceTag;
 		$urlProvider = new ConsulUrlProvider( $wgConsulUrl, $wgConsulServiceTag );
 		$apiProvider = new ApiProvider( $urlProvider );
-		return $apiProvider->getApi( self::SERVICE_NAME, TCSApi::class );
+		$apiClient = $apiProvider->getApi( self::SERVICE_NAME, TCSApi::class );
+
+		// default curl timeout for API client is set to 0 which = to no timeout.
+		// Overwriting to minimal value which is 1.
+		$apiClient->getConfig()->setCurlTimeout(1);
```

Comment:
Just for reference, this is the timeout for entire operation, including connection.